### ISSUE TITLE
[BUG] People hear their voice from stream

### ISSUE THREAD

**Issue by Tahinli** [Milkshiift/GoofCord#46]
### Description
People hear their voice from stream
### Steps to Reproduce
Start a stream
### Expected Behavior
I should be able to share my computer sound but people shouldn't hear discord voice
### Actual Behavior
Discord sound is also captured
### Environment
- Fedora 40 KDE
- GoofCord Version: 1.6.0 Flatpak

**Comment by cheshirecatt** [comment]
I also have the same problem in Windows 11

**Comment by MathiasDevelopes** [comment]
I have the same problem with flatpak and same version as op.

**Comment by Carthaigh-byte** [comment]
I experienced this as well today. Discord kept force closing while screen-sharing also.

**Comment by Wuemeli** [comment]
Does this only happen when screen sharing a Screen or an Application?

**Comment by Wuemeli** [comment]
@Milkshiift What does Goofcord use? venmic?

**Comment by Milkshiift** [comment]
No, just simple audio capture that Chromium provides. This issue is also present in ArmCord ([whatever it will be called when you read this](https://farside.link/nitter/ArmCordClient/status/1834617546445672703)), and Vesktop on Windows (because venmic is Linux only).

ArmCord recently made [a change](https://github.com/ArmCord/ArmCord/commit/dd4c0dcfdd5dd3f2c1f72ced2bef486c78942faf) that could potentially fix this on Windows, so if someone could test if this problem is present in the latest version of ArmCord on Windows it would be great.

**Comment by Wuemeli** [comment]
I will look into it

**Comment by ZachDavies** [comment]
Im still having the same issue on version 1.7.1

**Comment by Milkshiift** [comment]
Some info:
This is not getting fixed anytime soon on Windows. There is no Electron API to capture audio of specific windows. I'm guessing this could be possible with some native audio capturing implementation, akin to venmic but for Windows, but that is way out of scope.

Now for Linux, long-term, I plan to implement venmic sometime, though it seems to be a huge pain in the ass to do so. In the meantime, as a workaround, you can use qpwgraph to manually reroute needed audio to GoofCord capture.

**Comment by Tahinli** [comment]
> Some info: This is not getting fixed anytime soon on Windows. There is no Electron API to capture audio of specific windows. I'm guessing this could be possible with some native audio capturing implementation, akin to venmic but for Windows, but that is way out of scope.
> 
> Now for Linux, long-term, I plan to implement venmic sometime, though it seems to be a huge pain in the ass to do so. In the meantime, as a workaround, you can use qpwgraph to manually reroute needed audio to GoofCord capture.

I'm just curious about how Discord solves this problem, do you know what they do ?


In addition also you can try helvum to redirect sound sources.

**Comment by AyoungDukie** [comment]
I believe that discord installs its own driver/"audio helper"; in portable mode, it actually requires admin permissions to share audio (each time it's used the first time in a session I believe).

**Comment by 50keyboards-broken** [comment]
Hey, Has anyone else had this issue come up again? Im using Fedora 43 KDE 6.5.4 - Goofcord 2.0
People say they can hear their voice during screensharing.

**Comment by sluke0** [comment]
I signed back into Github just to mention that the most recent update for me has people hearing their own voice during screensharing too.

**Comment by Milkshiift** [comment]
Fixed in https://github.com/Milkshiift/GoofCord/commit/bf390e350637f4dfc85846cfd71f0f92d60381db
